src/primitives.jl

```diff
@@ -179,6 +181,30 @@ end
 dropnull(X::NullableVector) = X.values[!X.isnull]    # -> Vector
 
 """
+    dropnull!(X::AbstractVector)
+
+Remove null entries of `X` in-place and return the updated vector. This is a
+no-op if no nulls are present.
+
+    dropnull!(X::NullableVector)
+
+Remove null entries of `X` in-place and return a `Vector` view of `X.values`
+"""
+function dropnull!(X::AbstractVector)
+    if !(Nullable <: eltype(X))
+        return X
+    end
+    deleteat!(X, find(isnull, X))
```

Comment:
You're right of course, checking for missingness after removing missing values is kind of silly.

Though it looks like the two methods could easily be merged, with just a new branch like this at the top:
```julia
elseif T <: Nullable
    T = eltype(T)
end
```

Inference should be able to eliminate the `isa` check inside the loop.